### PR TITLE
Update Python to 3.11

### PR DESCRIPTION
**Pull request by brendanreardon**
In this pull request, we update MOAlmanac's Python version from 3.6 to 3.11. Python 3.8 was also tested to ensure backwards compatibility of outputs generated and required package versions. 

MD5 values did not match between produced outputs because the DataFrame.to_csv`lineterminator` changed in [Pandas 1.5.0](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_csv.html). Rather, files were individually inspected and content by element for all tab delimited outputs were compared using [pandas compare](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.compare.html). This was performed for all `config.ini` options being set both on and off and for Python versions 3.11 and 3.8. 

All unit tests pass. 

Revisions:
- Python was updated from 3.6 to 3.11
- Pandas was updated from 1.0.3 to 2.0.0
- Numpy was updated from 1.18.2 to 1.24.2
- Flask was updated from 2.0.0 to 2.0.3
- Requests was updated from 2.19.1 to 2.28.2
- Matplotlib was updated from 3.2.1 to 3.7.0
- Frozen Flask was updated from 0.15 to 0.18
- Scipy was updated from 1.4.1 to 1.10.1
- Scikit-learn was updated from 0.22.2 to 1.2.2
- ipykernel was updated from 5.1.0 to 6.22.0
- gdown, openpyxl, seaborn, tinydb, and xlrd were removed from `requirements.txt`. 
- The function "test_append_validation" was changed to fill NAs with empty strings rather than pd.NA as pd.NA [is still an experimental data type](https://pandas.pydata.org/docs/user_guide/missing_data.html). 
- `matchmaker.py` was revised to import `metrics` rather than `neighbors` from sklearn because sklearn.neighbors.DistanceMetric was moved to sklearn.metrics.DistanceMetric in 1.0. 
- the function `create_empty_output` within `matchmaker.py` was revised to change how columns are handled to be in line with Pandas best practices. 
- `illustrator.py` was revised to change behavior of setting tick labels and specifying a "tight" layout in accordance to updates to Matplotlib. 
- `requirements.txt` and `README.md` were updated to reflect these changes.

Bug fixes:
- A export statement was removed from `writer.txt` 

Pre-pull request check list,
- [x] Relevant documentation has been updated
- [x] All unit tests pass
- [x] All outputs pre- and post- changes match by md5 hash
- [x] All files changed have been reviewed

### CODE REVIEW

**Comment by brendanreardon** [comment]
All outputs match by md5. 
[main-branch-2023-04-13-all-off.zip](https://github.com/vanallenlab/moalmanac/files/11226049/main-branch-2023-04-13-all-off.zip)
[main-branch-2023-04-13.zip](https://github.com/vanallenlab/moalmanac/files/11226050/main-branch-2023-04-13.zip)
[upgrade-to-3.8.zip](https://github.com/vanallenlab/moalmanac/files/11226051/upgrade-to-3.8.zip)
[upgrade-to-3.11-all-off.zip](https://github.com/vanallenlab/moalmanac/files/11226052/upgrade-to-3.11-all-off.zip)
[upgrade-to-3.11.zip](https://github.com/vanallenlab/moalmanac/files/11226053/upgrade-to-3.11.zip)
